File: healthmate-backend/src/main/java/com/healthmate/service/WorkoutPlanService.java
package com.healthmate.service;

import com.healthmate.model.WorkoutPlan;
import com.healthmate.repository.WorkoutPlanRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class WorkoutPlanService {
    @Autowired
    private WorkoutPlanRepository workoutPlanRepository;

    public WorkoutPlan createWorkoutPlan(WorkoutPlan workoutPlan) {
        return workoutPlanRepository.save(workoutPlan);
    }

    public List<WorkoutPlan> getWorkoutPlansByTrainer(String trainerId) {
        return workoutPlanRepository.findByTrainerId(trainerId);
    }

    public void deleteWorkoutPlan(String id) {
        workoutPlanRepository.deleteById(id);
    }
}

File: healthmate-frontend/src/services/workoutPlan.service.js
import api from './api';

const createWorkoutPlan = (data) => {
    return api.post('/workout-plans', data);
};

const getWorkoutPlansByTrainer = (trainerId) => {
    return api.get(`/workout-plans/trainer/${trainerId}`);
};

const deleteWorkoutPlan = (id) => {
    return api.delete(`/workout-plans/${id}`);
};

const workoutPlanService = {
    createWorkoutPlan,
    getWorkoutPlansByTrainer,
    deleteWorkoutPlan
};

export default workoutPlanService;
